### PR TITLE
🐛 Add a broken unit test in 0.24.1

### PR DESCRIPTION
**Pull request by simkimsia**
Hi @fabiocaccamo 

The latest 0.24.1 broke something in my Django project.

I did a demo video showing you how I replicated the issue in a unit test. https://www.loom.com/share/1d50cce1764040658d9e9fe4747f9424

This fork also contains my unit test. Please feel free to go through.

Thank you

### CODE REVIEW

File: tests/dicts/test_assign.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        }
+        new_linkage = benedict({"list": {}})
+
+        output_config = required_fields_to_linkages["in_scope"]
+        output_key = output_config["output_key"]
+        new_linkage["list"][output_key] = {}
+
+        for output_field in output_config["fields"]:
+            new_linkage["list"][output_key][output_field] = None
+            self.assertTrue(True)
```
Comment (fabiocaccamo) [inline tests/dicts/test_assign.py:28]: How this could fail? :)

**Review by fabiocaccamo — CHANGES_REQUESTED** [review]

**Comment by simkimsia** [comment]
@fabiocaccamo did you watch my loom? It shows how when I run the same test under 0.24.0 it's fine but when i run 0.24.1 it fails.

**Comment by simkimsia** [comment]
@fabiocaccamo am i missing something? or u need more info from me? happy to make changes to the unit test to better flesh out the issue

**Comment by simkimsia** [comment]
I realized that the comment in https://github.com/fabiocaccamo/python-benedict/issues/59#issuecomment-892640386 has the same nature as my issue here. So making a comment to link both together

**Comment by simkimsia** [comment]
This is deleted after pr #60 is the better solution

**Pull request closed without merging**